crates/red_knot_python_semantic/resources/mdtest/annotations/annotated.md

```diff
@@ -0,0 +1,85 @@
+# `Annotated`
+
+`Annotated` attaches arbitrary metadata to a given type.
+
+## Usages
+
+`Annotated[T, ...]` is equivalent to `T`: All metadata arguments are simply ignored.
+
+```py
+from typing_extensions import Annotated
+
+def _(x: Annotated[int, "foo"]):
+    reveal_type(x)  # revealed: int
+
+def _(x: Annotated[int, lambda: 0 + 1 * 2 // 3, _(4)]):
+    reveal_type(x)  # revealed: int
+
+def _(x: Annotated[int, "arbitrary", "metadata", "elements", "are", "fine"]):
+    reveal_type(x)  # revealed: int
+
+def _(
+    x: Annotated[
+        tuple[str, int, tuple[bytes]],
+        # TODO: Emit an error here
+        (foo := b"SyntaxError: named expression cannot be used within an annotation"),
+    ]
+):
+    reveal_type(x)  # revealed: tuple[str, int, tuple[bytes]]
```

Comment:
Why should we emit an error on this? This works fine at runtime, on Python 3.9 and Python 3.13:

```pycon
>>> from typing import Annotated
>>> def f(x: Annotated[int, (foo := b"SyntaxError: named expression cannot be used within an annotation")]): ...
>>>
```

I'm not sure I understand what the value of this test is. If there's a syntax error that we're not detecting, that's either a bug in the parser or it's related to https://github.com/astral-sh/ruff/issues/11934. Neither is really something we need to test directly here. I think I would just delete this test; it seems confusing to me